test/JDBC/src/main/java/com/sqlsamples/JDBCTempTable.java

```diff
@@ -0,0 +1,339 @@
+        Statement s = c2.createStatement();
+
+        try {
+            for (int i = 0; i < 11; i++) {
+                String queryString = "CREATE TABLE #tab" + i + " (a int)";
+                s.execute(queryString);
+            }
+        } catch (Exception e) {
+            if (!e.getMessage().startsWith("Unable to allocate oid for temp table.")) {
+                bw.write(e.getMessage());
+                bw.newLine();
+            }
+        }
+
+        /* If the table was created, throw an error. */
+        ResultSet rs = s.executeQuery("SELECT * FROM babelfish_get_enr_list() WHERE relname = \'#table_cant_be_created\'");
+        if (rs.next()) {
+            bw.write("A table was created that should have reached buffer size limit.");
+            bw.newLine();
+        }
+
+        /*
+         * TEST: Ensure that we can wraparound properly.
+         */
+        rs = s.executeQuery("SELECT * FROM babelfish_get_enr_list() WHERE relname = \'#tab0\'");
+        if (!rs.next()) {
+            bw.write("Table is missing.");
+            bw.newLine();
+        }
+        int old_oid = Integer.parseInt(rs.getString("reloid"));
+        s.execute("DROP TABLE #tab0");
+        int new_oid = create_table_and_report_oid(c2, "CREATE TABLE #new_table(a int)", "#new_table");
+        
+        if (old_oid != new_oid) {
+            bw.write("Wraparound did not handle new OIDs properly.");
+            bw.newLine();
+        }
+
+        c2.close();
+
+        /* Restore GUC after tests. */
+        alter_guc.execute("ALTER DATABASE jdbc_testdb SET babelfishpg_tsql.temp_oid_buffer_size = 65536");
+        psql.close();
+        c.close();
+    }
+
+    /*
+     * Sanity checks that must be done from psql endpoint.
+     */
+    private static void psql_test(BufferedWriter bw, Logger logger) throws Exception {
+        /* Test GUC crash resiliency - Create a few normal tables (to advance OID counter). Ensure that the GUC remains the same from a different session. */
+        Connection connection = DriverManager.getConnection(connectionString);
+        Statement s = connection.createStatement();
+
+        /* Create a few tables so that the OID counter is not a predictable value. */
+        for (int i = 0; i < 100; i++)
+        {
+            s.execute("CREATE TABLE crash_res_dummy_table(a int)");
+            s.execute("DROP TABLE crash_res_dummy_table");
+        }
+        
+        s.execute("CREATE TABLE #t1(a int)");
+
+        /* Create two new connections to check the GUC value from postgres. */
+        Connection c = DriverManager.getConnection(connectionString);
+        JDBCCrossDialect cx = new JDBCCrossDialect(c);
+        Connection psql = cx.getPsqlConnection("-- psql", bw, logger);
+
+        Statement check_guc = psql.createStatement();
+        ResultSet rs = check_guc.executeQuery("SHOW babelfishpg_tsql.temp_oid_buffer_start");
+        if (!rs.next()) {
+            bw.write("Table is missing.");
```

Comment:
why table is missing message?

Reply:
I guess it's a slight misnomer, but we shouldn't expect this to ever fail, since it's just displaying a GUC value for us.